examples/js-fa2/src/fa2.ts

```diff
@@ -0,0 +1,376 @@
+  get_token_metadata: (token_id: token_id) => token_metadata
+  // returns the administrator of the fa2 storage
+  get_administrator: () => address | undefined
+  // add a token in the store
+  add_token: (token_id: token_id, token_metadata: token_metadata, amount: amount) => void;
+}
+
+// This function is used to create a fa2_storage (see the interface) form the Deku_storage
+// It's a kind of wrapper arround the deku_storage
+export const fa2_storage = (deku_storage: Deku_storage): fa2_storage => {
+  const get_token_amount = (address: address, token_id: token_id): amount => {
+    const json_str: string | undefined = deku_storage.get(`${address}-${token_id}`);
+    if (json_str === undefined) {
+      return 0 // If the address does not own the token, the amount of token is considered to be 0
+    }
+    const json: { operators: address[], amount: amount } = JSON.parse(json_str);
+    return json.amount
+  }
+
+  const exists = (token_id: token_id): boolean => {
+    const json_str = deku_storage.get(`${token_id}`);
+    return json_str !== undefined
+  }
+
+  const decrement_token_balance = (address: address, token_id: token_id, amount: amount) => {
+    const json_str = deku_storage.get(`${address}-${token_id}`);
+    if (json_str === undefined) {
+      throw FA2_Error.FA2_INSUFFICIENT_BALANCE
+    }
+    const json = JSON.parse(json_str);
+    const next = { ...json, amount: json.amount - amount };
+    deku_storage.set(`${address}-${token_id}`, JSON.stringify(next));
+  }
+
+  const increment_token_balance = (address: address, token_id: token_id, amount: amount) => {
+    const json_str = deku_storage.get(`${address}-${token_id}`);
+    const json: ledger_entry = json_str === undefined
+      ? { operators: [], amount: 0 }
+      : JSON.parse(json_str);
+    const next = { ...json, amount: json.amount + amount }
+    deku_storage.set(`${address}-${token_id}`, JSON.stringify(next));
+  }
+
+  const add_operator = (owner: address, token_id: token_id, operator: address) => {
+    const json_str = deku_storage.get(`${owner}-${token_id}`);
+    const json: ledger_entry = json_str === undefined
+      ? { amount: 0, operators: [] }
+      : JSON.parse(json_str)
+    const next = { ...json, operators: [...json.operators, operator] }
+    deku_storage.set(`${owner}-${token_id}`, JSON.stringify(next));
+  }
+
+  const remove_operator = (owner: address, token_id: token_id, operator: address) => {
+    const json_str = deku_storage.get(`${owner}-${token_id}`);
+    if (json_str === undefined) {
+      return;
+    } else {
+      const json: ledger_entry = JSON.parse(json_str);
+      const next = { ...json, operators: json.operators.filter(op => op !== operator) }
+      deku_storage.set(`${owner}-${token_id}`, JSON.stringify(next));
+    }
+  }
+
+  const get_operators = (address: address, token_id: token_id): address[] => {
+    const json_str = deku_storage.get(`${address}-${token_id}`);
+    if (json_str === undefined) {
+      return []
```

Comment:
I think `throw FA2_Error.FA2_TOKEN_UNDEFINED` is a better thing to do.

Reply:
In some case we don't want to throw exception, in this case the default value should be an empty array, you can see the [tzip-12 implementations](https://gitlab.com/tezos/tzip/-/blob/master/proposals/tzip-12/tzip-12.md) : 
> It is possible to update operators for a token owner that does not hold any token
balances yet.

So even if there isn't a token for a given user, you can interpret its operators as an empty array